org.uqbar.project.wollok/src/org/uqbar/project/wollok/interpreter/api/WollokInterpreterAccess.xtend

```diff
@@ -25,19 +26,7 @@ class WollokInterpreterAccess {
 	 * which is needed in different parts of the interpreter 
 	 */
 	def boolean wollokGreaterThan(Object a, Object b) {
-		operations.asBinaryOperation(">").apply(a, [|b]).isTrue()
+		operations.asBinaryOperation(">").apply(a, [|b]).isTrue
```

Comment:
It's not. Is just that now the contract of "asBinaryOperation" has been broaden so the function that it returns always a closure as second param. But most operations just return the same "b" object which is b already evaluated.
The only lazy ops are && and ||. The actual code that has makes it lazy is the calling method in the evaluator (similar to this one) but that is something like this
```
operations.asBinaryOperation("&&").apply(a, [| b.eval() ])
```